frontend/amundsen_application/static/js/features/ColumnList/index.tsx

```diff
@@ -272,6 +273,7 @@ const ColumnList: React.FC<ColumnListProps> = ({
       index,
     };
   };
+  const hideNestedColumns = getColumnCount(columns) >= getMaxNestedColumns();
```

Comment:
This might be costly performant wise, as it will be run on every render. If you could measure it that would be great. If that's the case, using useMemo will help here.

Reply:
Didn't know about this. Thanks for the tip!